scanners/tls.py

```diff
@@ -13,11 +16,46 @@
 # and only process domains with valid HTTPS, or broken chains.
 ###
 
-
 command = os.environ.get("SSLLABS_PATH", "ssllabs-scan")
-init = None
 workers = 5
 
+chrome_preload_list = None
+
+
+def get_chrome_preload_list():
+    logging.debug("Fetching chrome preload list...")
+
+    preload_list_url = ('https://chromium.googlesource.com/chromium/src/net/+'
+                        '/master/http/transport_security_state_static.json')
+    with urllib.request.urlopen(preload_list_url + '?format=text') as response:
+        raw = response.read()
+
+    # To avoid parsing the contents of the file out of the source tree viewer's
+    # HTML, we download it as a raw file. googlesource.com Base64-encodes the
+    # file to avoid potential content injection issues, so we need to decode it
+    # before using it. https://code.google.com/p/gitiles/issues/detail?id=7
+    raw = base64.b64decode(raw).decode('utf-8')
+
+    # The .json file contains '//' comments, which are not actually valid JSON,
+    # and confuse Python's JSON decoder. Begone, foul comments!
+    raw = ''.join([ re.sub(r'//.*$', '', line)
+                    for line in raw.splitlines() ])
+
+    preload_list_json = json.loads(raw)
+    return { entry['name'] for entry in preload_list_json['entries'] }
+
+
+def init(options):
+    """
+    Download the Chrome preload list at the beginning of the scan, and
+    re-use it for each scan. It is unnecessary to re-download the list for each
+    scan because it changes infrequently.
+    """
+    global chrome_preload_list
+    chrome_preload_list = get_chrome_preload_list()
+    return True
```

Comment:
This was a bit weird - I had to return a "truthy" value from the scanner's init function, or else `scan` would exit silently without any error or info message. This is due to the following code from `scan`:

```py
        # Scanners can have an optional init/validation hook.
        if scanner.init and (not scanner.init(options)):
            exit()
```

I'm fine with this, but it might be nice to document or explain what the intention is here. Would it be better to use exceptions instead of return values for communicating error conditions here?